test/data.jl

```diff
@@ -262,13 +262,13 @@ module TestData
 
     # # TODO: Restore this functionality
     # m1 = join(dt1, dt2, on = [:a,:b])
-    # @test isequal(m1[:a], Vector(["x", "x", "y", "y", fill("x", 5)]))
+    # @test m1[:a] == Vector(["x", "x", "y", "y", fill("x", 5)]))
```

Comment:
I just checked out your branches to test this out (sorry it took a while, I missed this notification!).

I wonder how long this has been commented out, and I have no idea what "functionality" we want to restore here. The above DataTables have columns of Symbols, not Strings as are being tested here. We also don't support calling column names as Strings, which is also done just below here with `m2 = join(dt1, dt2, on = ["a","b"], kind = :outer)`.

I updated this to be
```julia
m1 = join(dt1, dt2, on = [:a, :b])
@test m1[:a] == [:x, :x, :y, :y, :y, :x, :x, :x]
m2 = join(dt1, dt2, on = [:a, :b], kind = :outer)
@test m2[10,:v2] == null
@test m2[:a] == (?Symbol)[:x, :x, :y, :y, :y, :x, :x, :y, :x, :y, null, :y]
```
And the behavior and tests all look ok to me.